pkg/discovery/processor/controller_processor.go

```diff
@@ -130,3 +133,22 @@ func (cp *ControllerProcessor) cacheAllControllers() {
 	}
 	cp.KubeCluster.ControllerMap = controllerMap
 }
+
+func cacheController(obj unstructured.Unstructured) bool {
+	if obj.GetKind() != util.ReplicaSetResName &&
+		obj.GetKind() != util.ReplicationControllerResName {
+		return true
+	}
```

Comment:
Shall we also check `ownerReference`? That is, we only continue to check number of replicas when:
* this is a `ReplicaSet` and the owner is a `Deployment`, or
* this is a `ReplicationController` and the owner is a `DeploymentConfig`

Reply:
@ading1977 I think we also support the use case where a pod could be owned only by rs or rc  and not configured by a dep or dc (although almost nobody uses this).

Reply:
That is my understanding too. But with the current logic, if an rs or rc does not have owner, but has 0 replicas, then we won't be able to cache it. But in fact we do want to cache these controllers. Did I miss anything?